### PR TITLE
Added meaning full error message based on platform

### PR DESCRIPTION
**Pull request by Pankaj-SinghR**
Fixes https://github.com/element-hq/element-web/issues/25948
* Added deviceType condition check for web and desktop

* Added Error description for web and desktop

* Changed 'error_database_closed_title' Title to '%(brand)s stopped working'

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Added meaning full error message based on platform ([\#12074](https://github.com/matrix-org/matrix-react-sdk/pull/12074)). Fixes element-hq/element-web#25948. Contributed by @Pankaj-SinghR.

### CODE REVIEW

**Comment by Pankaj-SinghR** [comment]
If there are any more changes needed, please let me know. The coding standards have been updated since this issue was initially reported.

File: src/MatrixClientPeg.ts [hunk 1]
```diff
@@ -212,12 +213,22 @@ class MatrixClientPegClass implements IMatrixClientPeg {
...
             // For guests this is likely to happen during e-mail verification as part of registration
 
-            const { finished } = Modal.createDialog(ErrorDialog, {
-                title: _t("error_database_closed_title"),
-                description: _t("error_database_closed_description"),
-                button: _t("action|reload"),
-            });
-            const [reload] = await finished;
+            const ua = navigator.userAgent;
+            const deviceType = parseUserAgent(ua).deviceType;
```
Comment (t3chguy) [inline src/MatrixClientPeg.ts:217]: We shouldn't be relying on UA parsing, you can use the Platforms for this. WebPlatform vs ElectronPlatform.

Comment (Pankaj-SinghR) [reply src/MatrixClientPeg.ts]: > We shouldn't be relying on UA parsing, you can use the Platforms for this. WebPlatform vs ElectronPlatform.

I have made changes, PTAL 👍

**Comment by Pankaj-SinghR** [comment]
Just saw testcase was failing, pushed changes for testcase too. 👍

**Comment by t3chguy** [comment]
@Pankaj-SinghR seems the linter isn't happy with your change.

**Comment by Pankaj-SinghR** [comment]
> @Pankaj-SinghR seems the linter isn't happy with your change.

PTAL

**Review by t3chguy — APPROVED** [review]
Looks sane, thanks

**Pull request merged**